docker/k8s_tools.py

```diff
@@ -35,45 +35,70 @@ def wait_pods_running(label_selector, desired):
 
 
 def count_pods_by_phase(label_selector, phase):
-    pod_list = fetch_pods_info(label_selector)
-    filtered_pod_list = filter(lambda x: x[0] == phase, pod_list)
-    return len(filtered_pod_list)
+    pod_list = fetch_pods_info(label_selector, phase)
+    return len(pod_list)
 
 
-def fetch_pserver_ips():
-    label_selector = "paddle-job-pserver=%s" % PADDLE_JOB_NAME
-    pod_list = fetch_pods_info(label_selector)
-    pserver_ips = [item[1] for item in pod_list]
-    return ",".join(pserver_ips)
+def fetch_ips(label_selector, port=None, phase=None):
+    pod_list = fetch_pods_info(label_selector, phase)
+    ips = [item[1] for item in pod_list]
+    ips.sort()
 
+    r = []
+    for ip in ips:
+        if port != "0":
+            ip = "{0}:{1}".format(ip, port)
 
-def fetch_master_ip():
-    label_selector = "paddle-job-master=%s" % PADDLE_JOB_NAME
-    pod_list = fetch_pods_info(label_selector)
-    master_ips = [item[1] for item in pod_list]
-    return master_ips[0]
+        r.append(ip)
 
+    return ",".join(r)
 
-def fetch_trainer_id():
-    label_selector = "paddle-job=%s" % PADDLE_JOB_NAME
-    pod_list = fetch_pods_info(label_selector)
-    trainer_ips = [item[1] for item in pod_list]
-    trainer_ips.sort()
+
+def fetch_id(label_selector, phase=None):
+    pod_list = fetch_pods_info(label_selector, phase)
```

Comment:
These lines are the same as `fetch_ips`, try put them in one function.

Reply:
Done.